File: examples/databases/db-record-atomic.swift
import JsBaoClient

// Atomic direct-record operations: numeric increment/decrement and StringSet
// add/remove. These mutate without a read-modify-write round trip.
func recordAtomicOps(client: JsBaoClient, databaseId: String) async throws {
  let db = client.databases.connect(databaseId: databaseId)
  // #region example
  // Increment/decrement numeric fields; returns the post-increment values.
  let newValues = try await db.increment(
    modelName: "tasks", id: "task-1",
    fields: ["priority": 1, "estimatedHours": -2]
  )

  // Atomically add/remove StringSet members.
  try await db.addToSet(modelName: "tasks", id: "task-1", sets: ["tags": ["featured"]])
  try await db.removeFromSet(modelName: "tasks", id: "task-1", sets: ["tags": ["sale"]])
  // #endregion example
  _ = newValues
}

File: examples/databases/db-record-atomic.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Atomic direct-record operations: numeric increment/decrement and StringSet
// add/remove. These mutate without a read-modify-write round trip.
export async function recordAtomicOps(client: JsBaoClient, databaseId: string) {
  const db = client.databases.connect(databaseId);
  // #region example
  // Increment/decrement numeric fields; returns the post-increment values.
  const newValues = await db.increment("tasks", "task-1", {
    priority: 1,
    estimatedHours: -2,
  });

  // Atomically add/remove StringSet members.
  await db.addToSet("tasks", "task-1", { tags: ["featured"] });
  await db.removeFromSet("tasks", "task-1", { tags: ["sale"] });
  // #endregion example
  return newValues;
}
